### PR TITLE
Changing Asm parser

### PR DESCRIPTION
**Pull request by SeveroMorales**

### CODE REVIEW

**Comment by r0qs** [comment]
Hi, thank you for your contribution. Could you please describe what the PR does and what problem it solves? If it is related to an open issue, please add it as well.

**Comment by leonardoalt** [comment]
The PR seems to be allowing number literals with `_` between digits in Yul, which is already allowed in Solidity.
@SeveroMorales please confirm and link an issue if you've seen one somewhere.

File: libyul/AsmParser.cpp [hunk 1]
```diff
@@ -701,10 +701,19 @@ void Parser::checkBreakContinuePosition(string const& _which)
...
+		index = 0;
+		for(int i = 0; i < _literal.size(); i++)
+		{
+			if(_literal[i] != '_')
+			{
+				tmp_literal[index] = _literal[i];
+				index++;
+			}
+		}
+		[[maybe_unused]] auto tmp = u256(tmp_literal);
```
Comment (leonardoalt) [inline libyul/AsmParser.cpp:716]: May prefer `_literal.erase(std::remove(_literal.begin(), _literal.end(), '_'), _literal.end());`

**Comment by cameel** [comment]
So is it this issue: #13407?

If so, please add `Closes #13407` to the description and change the PR title to be a bit more descriptive.

File: libyul/AsmParser.cpp (hunk 1 above)
Comment (nikola-matic) [reply libyul/AsmParser.cpp:716]: Not may, but should definitely use this.

**Review by nikola-matic — CHANGES_REQUESTED** [review]
As others have mentioned (or not), please do the following:
1. Rename the PR title to something more specific, e.g. `Allow underscore separated number literals`
2. Add a description of what issue it closes; in this case, it closes (albeit partially, as you've not implemented support for scientific/short form notation, i.e 1e6) #13407
3. You'll need to add some test coverage, either in `test/libsolidity/semanticTests/literals/*` or `test/libyul/*` (someone else can chime in which would be the appropriate place, as we don't really have much coverage for number literals in `assembly` block from what I can tell)

**Comment by nikola-matic** [comment]
@SeveroMorales do you plan on continuing with this PR?

**Comment by nikola-matic** [comment]
@SeveroMorales I'm closing this PR due to inactivity. Thanks for the contribution, and in the future, please either describe the reason for the PR in the description text box, or even better, link an issue you're trying to fix. If no such issue exists, please first create an issue and leave some time for discussion, and only then open a PR.

If you'd like to continue on this, feel free to reopen the PR and ping us for help if you need any :)

**Pull request closed without merging**